### ISSUE TITLE
[Feature Request] Automatically reload auth file

### ISSUE THREAD

**Issue by penguinpowernz** [jpillora/chisel#30]
This feature assumes connections will be broken if server is restarted.

This would come in handy for building the auth file programmatically from a separate data source to avoid.

There's three ways we could do it in increasing levels of complexity:

1. Reload every x second regardless
2. Catch SIGUSR1 to trigger a reload
3. Use inotify to trigger a reload when the file changes

Not sure which will be best.

**Comment by jpillora** [comment]
I think a poll every 5 seconds would be simplest. It's not a huge change, would you willing to send a PR?

This from `server.go`, can be moved into a function
```go
		users, err := chshare.ParseUsers(config.AuthFile)
		if err != nil {
			return nil, err
		}
		s.Users = users
```

The function can run in a loop on a timer, and `s.Users` will need a `sync.Mutex` protecting it

**Comment by penguinpowernz** [comment]
Yea mate, I'll take a stab at it. Cheers

**Comment by jpillora** [comment]
This has been implemented in 1.2.4
